### PR TITLE
Feature: implement base field component architecture with text field implementation

### LINKED ISSUE #211: Implement components for task template integration
## Context
Task templates in our application need to support dynamic form fields with various input types and constraints. We need a flexible, reusable component system that can handle different field types while maintaining consistent behavior for validation, editing modes, and user interactions across the entire application.

### 1. Base Architecture
- Create a generic base field component that provides common functionality for all field types:
  - Label and description rendering
  - Validation error display
  - Constraint hints
  - Mode switching (edit/view/toggleable modes)
  - State management
- Define a flexible interaction mode system to support different use cases:
  - **EditOnly**: Direct editing with immediate value updates
  - **ViewOnly**: Read-only display
  - **Toggleable**: Buffered editing with save/cancel actions

### 2. Field Component Implementations
Implement the following field components that extend the base architecture:

- **TextFieldComponent**: Text input with character limits, placeholders, and multi-line support
- **NumberFieldComponent**: Numeric input with min/max constraints, decimal support, and optional unit display
- **DateFieldComponent**: Date selection with optional time support and date range constraints
- **SingleSelectFieldComponent**: Dropdown selection with human-readable labels, value mapping, and custom value input
- **MultiSelectFieldComponent**: Multi-selection with visual feedback, min/max selection constraints, and custom value support

### 3. Design Principles
- Components must be reusable and adaptable for different templates
- Consistent UI/UX across all field types
- Type-safe implementation using Kotlin generics
- Comprehensive validation support
- All components should be well-tested with unit tests covering interaction modes, validation, constraints, and edge cases

## Acceptance Criteria
- [ ] Base field component architecture is implemented with mode management
- [ ] All five field types are implemented and functional
- [ ] Components work in all three interaction modes
- [ ] Validation and constraints are properly enforced
- [ ] Comprehensive test coverage for all components
- [ ] Components are reusable for other forms beyond task templates

---

**Note:** This work is split across PR #264 (base architecture + text field) and PR #265 (remaining field components).

This message was generated with Claude Code

### PR DESCRIPTION
**Pull request by ahlombardini**
## Context
Task templates in our application require dynamic form fields with various input types and constraints. To support this functionality, we need a flexible, reusable component system that can handle different field types while maintaining consistent behavior for validation, editing modes, and user interactions.

## Summary
This PR introduces the foundational architecture for template field components:
- **BaseFieldComponent**: Generic wrapper component providing layout, validation, mode switching, and state management for all field types
- **FieldInteractionMode**: Sealed interface defining three interaction patterns (EditOnly, ViewOnly, Toggleable)
- **TextFieldComponent**: Text input field with character limits, placeholders, and multi-line support
- **ExcludeFromJacocoGeneratedReport**: Annotation to exclude preview functions from code coverage
- **Comprehensive test coverage**: BaseFieldComponentTest, TextFieldComponentTest, BaseFieldComponentLogicTest, and FieldInteractionModeTest validate all modes, validation behavior, and edge cases

## Why these changes
The implementation uses a generic architecture with Kotlin type parameters to create flexible, reusable components. BaseFieldComponent handles all common functionality (labels, descriptions, validation errors, constraint hints, mode toggle buttons) through a generic `renderer` lambda pattern. This allows field-specific implementations to focus solely on their input UI while inheriting all shared behavior.

The three interaction modes support different use cases:
- **EditOnly**: Direct editing with immediate value updates
- **ViewOnly**: Read-only display
- **Toggleable**: Buffered editing with save/cancel actions

TextFieldComponent provides a concrete example of how to build on this foundation, handling text input with character limits, placeholders, and multi-line support.

## Potential future improvements
This architecture is designed to be extended to additional field types (Number, Date, SingleSelect, MultiSelect) and could be leveraged for all forms throughout the application beyond task templates.

## Potential future bugs
- The generic type system may require adjustments as more complex field types are added
- State management between modes might need refinement based on real-world usage patterns

---
*Note: This PR description was generated with [Claude Code](https://claude.com/claude-code) assistance.*

Closes #211

### CODE REVIEW

File: app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/BaseFieldComponent.kt [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+      remember(mode) {
+        mutableStateOf(
+            if (mode is FieldInteractionMode.Toggleable && mode.isEditing) value else null)
+      }
+
+  var prevIsEditing by remember { mutableStateOf(mode.isEditing) }
+
+  LaunchedEffect(mode.isEditing) {
+    if (mode.isEditing && !prevIsEditing && mode is FieldInteractionMode.Toggleable) {
+      originalValue = value
```
Comment (Tuaer) [inline app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/BaseFieldComponent.kt]: For me, you are passing a reference here, which is a bit risky, because some other code can modify the referenced value. Wouldn't it be better to perform a copy?

**Comment by Tuaer** [comment]
There are also 2 issues that Sonar found. They are related to the complexity. Try to fix them, but if this turns out to be too complicated, I can approve without it.

File: app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/BaseFieldComponent.kt (hunk 1 above)
Comment (ahlombardini) [reply app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/BaseFieldComponent.kt]: We are going to assume that the user won't make weird manipulations as it is impossible to leverage data classes and serialization at the same time in this case.
